### PR TITLE
feat: enhance gemini models

### LINKED ISSUE #9273: Concurrent API requests to Gemini vision model cause non-responsive behavior and block all tasks forever
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.9.1 / 0.9.2 / 0.11.2

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

1.  Configure Gemini  1.5  Pro 001
2.  Create  a  new workflow  using  LLM  with  visual  capabilities to  briefly  analyze  images
3.  Upload  an  image  via  API,  then  make  50  concurrent  API calls  to  execute  the image  analysis  workflow (I m using java, and 20 concurrent calls is fine). The  API  calls  will hang  indefinitely  without returning,  and  all LLM  operations on  the  site  will become  blocked
image:
31z2b5uzm9

4.  According  to the  workflow  logs, LLM  requests are  successfully  returned
 image

5. This  issue  does not  occur  with  GPT vision model

### ✔️ Expected Behavior

Results should be returned normally

### ❌ Actual Behavior

Results are not returned as expected, and the entire site becomes globally blocked. The site can only be restored by restarting

### LINKED ISSUE #10720: LLM video understanding
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Version 0.11.1 adds support for LLM video understanding. According to #9828, qwen mode requires users to submit a work order application and only supports urlsend mode. Please tell me how to do this and where to submit a work order?
In addition, what model is used to test qwen that supports video understanding?

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by hjlarry**
# Summary

> [!Tip]
> Close issue syntax: `Fixes #<issue number>` or `Resolves #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

1. all the google models support video file.  can close https://github.com/langgenius/dify/issues/10720 . maybe resolve https://github.com/langgenius/dify/issues/9273
2. use google's upload file API instead of base64 string.  Now we can cache the uploaded file instead of transfer base64 string each talk. Google support 1 hour video, transfer file each time is so terriable.
3. "gemini-pro-vision" was deprecated, remove this part of logic.
4. make the `DocumentPromptMessageContent` `ImagePromptMessageContent` `VideoPromptMessageContent` and `AudioPromptMessageContent` unified, both include `format` and  contains `mime_type` in their `data`.
5. please help test pdf of `claude` and audio of `openai`

# Screenshots

52f3a45d3c3cfb71be7e93fe16f340b

# Checklist

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

### CODE REVIEW

File: api/core/file/file_manager.py [hunk 1]
```diff
@@ -50,12 +50,12 @@ def to_prompt_message_content(
             else:
                 data = _to_base64_data_string(f)
 
-            return ImagePromptMessageContent(data=data, detail=image_detail_config)
+            return ImagePromptMessageContent(data=data, detail=image_detail_config, format=f.extension.lstrip("."))
```
Comment (laipz8200) [inline api/core/file/file_manager.py:53]: maybe `removeprefix()` is better?

File: api/core/model_runtime/model_providers/anthropic/llm/llm.py [hunk 2]
```diff
@@ -526,17 +526,19 @@ def _convert_prompt_messages(self, prompt_messages: Sequence[PromptMessage]) ->
                                 }
                                 sub_messages.append(sub_message_dict)
                             elif isinstance(message_content, DocumentPromptMessageContent):
-                                if message_content.mime_type != "application/pdf":
+                                data_split = message_content.data.split(";base64,")
+                                mime_type = data_split[0].replace("data:", "")
+                                base64_data = data_split[1]
```
Comment (laipz8200) [inline api/core/model_runtime/model_providers/anthropic/llm/llm.py:531]: I think we need some helpers or properties in PromptMessageContent model to extract mime_type and b64_data.

**Review by laipz8200 — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/anthropic/llm/llm.py (hunk 2 above)
Comment (hjlarry) [reply api/core/model_runtime/model_providers/anthropic/llm/llm.py:531]: sounds good !  It's a better way!

Comment (hjlarry) [reply api/core/model_runtime/model_providers/anthropic/llm/llm.py:531]: I make changes in another PR https://github.com/langgenius/dify/pull/11721 , this PR plan to only modify related with gemini models

**Review by laipz8200 — APPROVED** [review]

**Pull request merged**